gradle-plugins/biz.aQute.bnd.gradle/src/main/java/aQute/bnd/gradle/BundleTaskExtension.java

```diff
@@ -242,6 +250,25 @@ public BundleTaskExtension(org.gradle.api.tasks.bundling.Jar task) {
 			.property("default Bundle-Version", getDefaultBundleVersion());
 	}
 
+	/**
+	 * Adds an {@link ArtifactView} to {@link #classpath(Object...)} that will explicitly request variants of dependencies with the
+	 * {@link LibraryElements#JAR} library element set to {@link LibraryElements#JAR}.
+	 *
+	 * This method ensures that Gradle does not perform an optimization to local project dependencies and request the
+	 * {@link LibraryElements#CLASSES} value instead.  Doing so would add the directory containing the other project's compiled classes
+	 * to the classpath instead of the jar itself.  If the jar is not present, reading information from the {@code MANIFEST.MF}
+	 * file will not be possible, so we want to avoid Gradle applying this optimization.
+	 *
+	 * @param mainCompileClasspath the main source set of the project
+	 * @param objects the object factory of the project
+	 */
+	private void requestJarLibrary(Configuration mainCompileClasspath, ObjectFactory objects) {
+		ArtifactView jarView = mainCompileClasspath.getIncoming().artifactView(viewConfiguration -> {
+			viewConfiguration.attributes(attributeContainer -> attributeContainer.attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements.class, LibraryElements.JAR)));
+		});
+		classpath(jarView.getFiles());
```

Comment:
A slightly different variation on this would be to return the FileCollection and put the `classpath(requestJarLibrary(...))` in the caller.

Reply:
Yep.  I'm not sure which would read clearer.  I'll leave it to the project maintainers to choose.